prov/efa/src/rxr/rxr_msg.c

```diff
@@ -93,10 +97,40 @@ ssize_t rxr_msg_post_rtm(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_entry)
 		return rxr_pkt_post_ctrl_or_queue(rxr_ep, RXR_TX_ENTRY, tx_entry, rtm_type + tagged, 0);
 	}
 
+	if (delivery_complete_requested && !(peer->flags & RXR_PEER_HANDSHAKE_RECEIVED)) {
+		/*
+		 * Because delivery complete is defined as an extra
+		 * feature, the receiver might not support it.
+		 *
+		 * The sender cannot send with FI_DELIVERY_COMPLETE
+		 * if the peer is not able to handle it.
+		 *
+		 * If the sender does not know whether the peer
+		 * can handle it, it needs to wait for
+		 * a handshake packet from the peer.
+		 *
+		 * The handshake packet contains
+		 * the information whether the peer
+		 * support it or not.
+		 */
+
+		err = rxr_pkt_wait_handshake(rxr_ep, tx_entry->addr, peer);
```

Comment:
Can we not use our existing handshake used for other feature negotiation instead of posting one here?

Reply:
I used `rxr_pkt_wait_handshake` because the receiver will post handshake after it receives the first packet from the sender. Which means that if we do not have an extra packet for triggering a handshake, at least one packet will be send without using the DC protocol.

Reply:
That makes sense. It might have been better to extend the existing handshake's send-side logic to either block for a receipt (which implicitly gives the first message delivery complete semantics) or asynchronously negotiate features in other cases; but I'm fine with changing that after this PR too. We will inevitably have other optional features that might need such a blocking handshake during negotiation.